### PR TITLE
Vet clinic Milestone 4

### PR DESCRIPTION
**Pull request by ambrose-kibet**
## In this Milestone we:
- [ ] Added data for animals, vets, and visits.
- [ ] Ran queries to answer specific questions about the data, 
query1
query2
query3
query4
query5
query6
query7
query8
query9

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -86,4 +86,64 @@ SELECT owners.full_name, COUNT(animals.id) AS num_animals_owned
...
+SELECT animals.name 
+FROM animals 
+JOIN visits ON animals.id = visits.animal_id 
+JOIN vets ON vets.id = visits.vet_id 
+WHERE vets.name = 'William Tatcher' 
+ORDER BY visits.visit_date DESC 
+LIMIT 1;
+
+SELECT COUNT(DISTINCT animal_id) FROM visits 
+WHERE vet_id = (SELECT id FROM vets WHERE name = 'Stephanie Mendez');
```
Comment (Felix45) [inline queries.sql:100]: - [ Required ] I think this query will not give you the _Different animals Stephanie Mendez saw_, to do that you need to create joins between the `animals`, `visits`, and `vets` tables. You can do this by refactoring your query as shown below.

```ruby
  SELECT DISTINCT animals.name
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vets.name = 'Stephanie Mendez';
```

File: queries.sql [hunk 2]
```diff
@@ -86,4 +86,64 @@ SELECT owners.full_name, COUNT(animals.id) AS num_animals_owned
...
+
+SELECT vets.name,specializations.species_id AS specialization_species_id FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+ORDER BY vets.name;
```
Comment (Felix45) [inline queries.sql:104]: - [ Required ] I think this query will not give you the _List all vets and their specialties, including vets with no specialties_, to do that you need to create joins between `vets`, `specializations`, and `species` tables. You can do this by refactoring your query as shown below.

```ruby
  SELECT vets.*, species.name
  FROM vets
  LEFT JOIN specializations ON vets.id = specializations.vet_id
  LEFT JOIN species ON specializations.species_id = species.id;
```

File: queries.sql [hunk 3]
```diff
@@ -86,4 +86,64 @@ SELECT owners.full_name, COUNT(animals.id) AS num_animals_owned
...
+LEFT JOIN specializations vs ON vt.id = vs.vet_id AND a.species_id = vs.species_id
+WHERE vs.species_id IS NULL;
+
+SELECT animals.species_id, COUNT(*) AS num_visits
+FROM visits
+JOIN animals ON visits.animal_id = animals.id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Maisy Smith'
+GROUP BY animals.species_id
+ORDER BY num_visits DESC;
```
Comment (Felix45) [inline queries.sql]: - [ Required ] I think this query will not give the name of the species that answers the question _What specialty should Maisy Smith consider getting? Look for the species she gets the most_, to do that you need to create joins between `animals`, `vets`, and `species` tables. Currently, your query returns an `Id` which means someone still has to refer to the `species` table to get the `name` of the species. You can do this by refactoring your query as shown below.

```ruby
  SELECT species.name, count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN species ON animals.species_id = species.id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vet.name = 'Maisy Smith'
  GROUP BY species.name

```

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @ambrose-kibet, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [x] Good SQL query design :white_check_mark: 
- [x] Descriptive pull request title and summary :white_check_mark: 
- [x] Gitflow is used correctly :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:    

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (ambrose-kibet) [reply queries.sql:104]: @Felix45 This query returns exactly   List all vets and their specialties, including vets with no specialties as in the screenshot   the * wildcard adds unnecessary information about the vets

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @ambrose-kibet       , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] linters are passing
- [x] Professional README 💯 
- [x] Database functioning as intended 🥇 
 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**